frontend/src/core/comments/components/AddComment.js

```diff
@@ -1,97 +1,393 @@
+        () => withMentions(withReact(createEditor())), []
+    );
+    const initialValue = [{ type: 'paragraph', children: [{ text: '' }] }];
+    const [ value, setValue ] = React.useState(initialValue);
 
-    const commentInput: any = React.useRef();
-    const minRows = 1;
-    const maxRows = 6;
+    // Set focus on Editor
+    React.useEffect(() => {
+        if (!parameters || parameters.project !== 'terminology') {
+            ReactEditor.focus(editor);
+            Transforms.select(editor, Editor.end(editor, []));
+        }
+    }, [editor, parameters])
+    
+    React.useEffect(() => {
+        getUsers();
+    }, [getUsers])
 
-    if (!user) {
-        return null;
-    }
+    const usersList = users.users;
+    const USERS = usersList.map(user => user.name);
+    const chars = USERS.filter(c =>
+        c.toLowerCase().startsWith(search.toLowerCase())
+    ).slice(0, 5);
+
+    // Set position of mentions suggestions
+    React.useLayoutEffect(() => {
+        if (!target || chars.length <= 0) {
+            return;
+        }
+        const el = mentionList.current;
+        const domRange = ReactEditor.toDOMRange(editor, target);
+        const rect = domRange.getBoundingClientRect();
+        const teamCommentsEl = document.querySelector('.top');
+        const teamCommentsRect = !teamCommentsEl ? null : teamCommentsEl.getBoundingClientRect();
+        const teamCommentsActive = !teamCommentsEl ? false : teamCommentsEl.contains(document.activeElement);
+        const translateCommentsEl = document.querySelector('.history');
+        const translateCommentsRect = !translateCommentsEl ? null : translateCommentsEl.getBoundingClientRect();
+        const translateCommentsActive = !translateCommentsEl ? false : 
+            translateCommentsEl.contains(document.activeElement);
+        const editorMenu = document.querySelector('.editor-menu');
+        const editorMenuHeight = !editorMenu ? 0 : editorMenu.clientHeight;
+        const tabIndex = document.querySelector('.react-tabs__tab-list');
+        const tabIndexHeight = !tabIndex ? 0 : tabIndex.clientHeight;
+        const commentEditor = document.querySelector('.comments-list .add-comment .comment-editor');
+        const commentEditorLineHeight = parseInt(window.getComputedStyle(commentEditor).lineHeight) || 0;
+        const commentEditorTopPadding = parseInt(window.getComputedStyle(commentEditor).paddingTop) || 0;
+        const commentEditorBottomPadding = parseInt(window.getComputedStyle(commentEditor).paddingBottom) || 0;
+        const commentEditorSpan = document.querySelector('.comments-list .add-comment .comment-editor p span');
+        const commentEditorSpanHeight = !commentEditorSpan ? 0 : commentEditorSpan.offsetHeight;
+        const setTopAdjustment = commentEditorBottomPadding + commentEditorSpanHeight;
+        const suggestionsHeightAdjustment = (
+            commentEditorTopPadding + ((commentEditorLineHeight - commentEditorSpanHeight) / 2));
```

Comment:
This whole function, and this block in particular, are quite complex. I wished this was handled by an external library, but I assume that might not be a possible option here? 

If so, then for the sake of maintainability (and the sanity of developers, including you, who might need to look back at this code in the future :wink: ), I would highly recommend to take special care to make this as easy to read and understand as possible. That can mean comments, logical blocks of lines, variable names, and probably a combination of all those. If the complexity of the code cannot be lowered, then the number of comments needs to be much higher, as it is easier to read English than to read code.